polaris/benchmark/_base.py

```diff
@@ -147,9 +147,21 @@ def _validate_metrics(cls, v):
         if not isinstance(v, list):
             v = [v]
 
-        v = [m if isinstance(m, Metric) else Metric[m] for m in v]
+        tmp = []
+        unique_metrics = set()
+        for m in v:
+            if isinstance(m, str):
+                m = Metric[m]
 
-        if len(set(v)) != len(v):
+            if isinstance(m, Metric):
+                unique_metrics.add(m.name)
+            else:
+                unique_metrics.add(m.metric.name)
+
+            tmp.append(m)
+        v = tmp
+
+        if len(unique_metrics) != len(v):
             raise InvalidBenchmarkError("The task specifies duplicate metrics")
 
         if len(v) == 0:
```

Comment:
**Suggestion**
Something a little more compact:

```python
unique_metrics = set(Metric[m] if isinstance(m, str) else m for m in v)

if len(unique_metrics) != len(v):
    raise InvalidBenchmarkError("The task specifies duplicate metrics")

if len(unique_metrics) == 0:
    raise InvalidBenchmarkError("Specify at least one metric")

return unique_metrics
```

Reply:
Nice! For my own understanding: This does require [your other suggestion](https://github.com/polaris-hub/polaris/pull/223#discussion_r1858764073) to work, right?

Reply:
Yep.